src/main/scala/org/encryfoundation/generator/actors/BoxesHolder.scala

```diff
@@ -0,0 +1,51 @@
+package org.encryfoundation.generator.actors
+
+import akka.actor.{Actor, Cancellable, Props}
+import com.typesafe.scalalogging.StrictLogging
+import org.encryfoundation.generator.actors.BoxesHolder.{AskBoxesFromGenerator, BoxesAnswerToGenerator, BoxesRequestFromLocal}
+import org.encryfoundation.generator.transaction.box.EncryBaseBox
+import org.encryfoundation.generator.utils.Settings
+import org.encryfoundation.generator.wallet.{WalletStorage, WalletStorageReader}
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+
+class BoxesHolder(settings: Settings, walletStorageReader: WalletStorageReader) extends Actor with StrictLogging {
+
+  var walletStorage: WalletStorage = walletStorageReader.createWalletStorage
+
+  val defaultAskTime: FiniteDuration = settings.boxesHolderSettings.askBoxesFromLocalDBPeriod.seconds
+
+  val takeBoxesSchedule: Cancellable = context.system.scheduler.schedule(
+    5.seconds,
+    settings.boxesHolderSettings.askBoxesFromLocalDBPeriod.seconds,
+    self,
+    BoxesRequestFromLocal
+  )
+
+  override def receive: Receive = boxesHolderBehavior()
+
+  def boxesHolderBehavior(boxes: List[EncryBaseBox] = List()): Receive = {
+    case BoxesRequestFromLocal =>
```

Comment:
```suggestion
    case BoxesRequestFromLocal => context.become(boxesHolderBehavior(walletStorageReader.createWalletStorage.allBoxes.toList))
```
By doing this we can eliminate need of ```walletStorage``` variable